mmv1/third_party/terraform/resources/resource_apigee_shared_flow.go

```diff
@@ -0,0 +1,439 @@
+					},
+				},
+			},
+			"revision": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: `A list of revisions of this shared flow.`,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"config_bundle": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `A path to the config bundle zip you want to upload. Must be defined if content is not.`,
+			},
+			"md5hash": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Base 64 MD5 hash of the uploaded config bundle.`,
+			},
+			"detect_md5hash": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "Different Hash",
+				Description: `A hash of local config bundle in string, user needs to use a Terraform Hash function of their choice. A change in hash will trigger an update.`,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					localMd5Hash := ""
+					if config_bundle, ok := d.GetOkExists("config_bundle"); ok {
+						localMd5Hash = getFileMd5Hash(config_bundle.(string))
+					}
+					if localMd5Hash == "" {
+						return false
+					}
+
+					// `old` is the md5 hash we speculated from server responses,
+					// when apply responded with succeed, hash is set to the hash of uploaded bundle
+					if old != localMd5Hash {
+						return false
+					}
+
+					return true
+				},
+			},
+		},
+		UseJSONNumber: true,
+	}
+}
+
+func resourceApigeeSharedFlowCreate(d *schema.ResourceData, meta interface{}) error {
+	ctx := context.TODO()
+	tflog.Info(ctx, "resourceApigeeSharedFlowCreate")
+	log.Printf("[DEBUG] resourceApigeeSharedFlowCreate")
+
+	//print debug info======
+	tmpName, _ := d.GetOk("name")
+	tmpOrgId, _ := d.GetOk("org_id")
+	tmpBundle, _ := d.GetOk("config_bundle")
+	
+	log.Printf("[DEBUG] resourceApigeeSharedFlowCreate, name=			 	%s", tmpName)
+	log.Printf("[DEBUG] resourceApigeeSharedFlowCreate, org_id=, 			%s", tmpOrgId)
+	log.Printf("[DEBUG] resourceApigeeSharedFlowCreate, config_bundle=, 	%s", tmpBundle)
+	//debug=================
+
+	config := meta.(*Config)
+	userAgent, err := generateUserAgentString(d, config.userAgent)
+	if err != nil {
+		return err
+	}
+
+	var file *os.File
+	var localMd5Hash string
+	if configBundlePath, ok := d.GetOk("config_bundle"); ok {
+		var err error
+		file, err = os.Open(configBundlePath.(string))
+		if err != nil {
+			return err
+		}
+		localMd5Hash = getFileMd5Hash(configBundlePath.(string))
+	} else {
+		return fmt.Errorf("Error, \"config_bundle\" must be specified")
+	}
+
+	url, err := replaceVars(d, config, "{{ApigeeBasePath}}organizations/{{org_id}}/sharedflows?name={{name}}&action=import")
+	if err != nil {
+		return err
+	}
+	billingProject := ""
+
+	// err == nil indicates that the billing_project value was found
+	if bp, err := getBillingProject(d, config); err == nil {
+		billingProject = bp
+	}
+
+	log.Printf("[DEBUG] resourceApigeeSharedFlowCreate, url=, 	%s", url)
```

Comment:
nit: can we make the log lines sentences? just to be consistent with our current logs. for example `fmt.Sprintf("created apigee shared flow: %s", url)`. it's not a big deal, i just think that will be a bit more consistent with our current logs. feel free to look around the codebase for examples of what is being logged etc.

this applies to any of the logs in this file, i won't flag all of them. thanks!